From: Joshua Colp <jcolp@digium.com>
Subject: SRTP Video Remote Crash Vulnerability
 Prevent crash when an SDP offer is received with an encrypted video stream
 when support for video is disabled and res_srtp is loaded.
Origin: upstream, http://svnview.digium.com/svn/asterisk?view=revision&revision=351504
Bug: https://issues.asterisk.org/jira/browse/ASTERISK-19202
--- a/channels/chan_sip.c
+++ b/channels/chan_sip.c
@@ -28838,6 +28838,12 @@ static int setup_srtp(struct sip_srtp **
 
 static int process_crypto(struct sip_pvt *p, struct ast_rtp_instance *rtp, struct sip_srtp **srtp, const char *a)
 {
+	/* If no RTP instance exists for this media stream don't bother processing the crypto line */
+	if (!rtp) {
+		ast_debug(3, "Received offer with crypto line for media stream that is not enabled\n");
+		return FALSE;
+	}
+
 	if (strncasecmp(a, "crypto:", 7)) {
 		return FALSE;
 	}
